### ISSUE TITLE
Has anyone implemented retrieval model in ways other than 2 tower model?

### ISSUE THREAD

**Issue by amit-timalsina** [tensorflow/recommenders#436]
Like for this implementation, how can we use TensorFlow recommenders?
![image](https://user-images.githubusercontent.com/30175128/149714833-46736a13-ad31-458b-8382-897f57ad22d8.png)

**Comment by patrickorlando** [comment]
hey @amit-timalsina, the picture above describes the serving case for a two tower retrieval model.

**Comment by amit-timalsina** [comment]
> hey @amit-timalsina, the picture above describes the serving case for a two tower retrieval model.

Do you mean those two embedded video watches and embedded search tokens tower are two towers?

**Comment by AzizIlyosov** [comment]
@amit-timalsina I guess that  model is  implemented in  softmax  model section of  this  [colab](https://colab.research.google.com/github/google/eng-edu/blob/main/ml/recommendation-systems/recommendation-systems.ipynb?utm_source=ss-recommendation-systems&utm_campaign=colab-external&utm_medium=referral&utm_content=recommendation-systems#scrollTo=4OKzworlo4ue)

**Comment by maciejkula** [comment]
The softmax model you picture here is a degenerate case of a two-tower model. The softmax is exactly equivalent to having a candidate tower with a single embedding layer, with one embedding vector per candidate.
